go/tasks/plugins/array/k8s/monitor.go

```diff
@@ -91,6 +88,22 @@ func LaunchAndCheckSubTasksState(ctx context.Context, tCtx core.TaskExecutionCon
 		return currentState, logLinks, subTaskIDs, err
 	}
 
+	// identify max parallelism
+	taskTemplate, err := tCtx.TaskReader().Read(ctx)
```

Comment:
So this function gets called every time we check for task status... I would try to avoid rereading the task template over and over again... any chance we can do away with that (in a backward compatible way too)?

Reply:
@EngHabu so the task reader is cached. You are worried about a bad pattern?

Reply:
aha... I've not noticed that this block existed but just moved around... It looks like it was a remnant of some refactoring this went through...

I was think of moving it to task state but then we will have to write code for migration and that's not ideal either... 🤷🏽